homeassistant/components/here_travel_time/config_flow.py

```diff
@@ -150,37 +164,49 @@ async def async_step_origin_coordinates(
             self._config[CONF_ORIGIN_LONGITUDE] = user_input[CONF_ORIGIN][
                 CONF_LONGITUDE
             ]
+            self._config.pop(CONF_ORIGIN_ENTITY_ID, None)
             return await self.async_step_destination_menu()
         schema = vol.Schema(
             {
                 vol.Required(
                     CONF_ORIGIN,
                     default={
-                        CONF_LATITUDE: self.hass.config.latitude,
-                        CONF_LONGITUDE: self.hass.config.longitude,
+                        CONF_LATITUDE: self._config.get(CONF_ORIGIN_LATITUDE)
+                        or self.hass.config.latitude,
+                        CONF_LONGITUDE: self._config.get(CONF_ORIGIN_LONGITUDE)
+                        or self.hass.config.longitude,
                     },
                 ): LocationSelector()
             }
         )
         return self.async_show_form(step_id="origin_coordinates", data_schema=schema)
 
-    async def async_step_destination_menu(self, _: None = None) -> ConfigFlowResult:
-        """Show the destination menu."""
-        return self.async_show_menu(
-            step_id="destination_menu",
-            menu_options=["destination_coordinates", "destination_entity"],
-        )
-
     async def async_step_origin_entity(
         self, user_input: dict[str, Any] | None = None
     ) -> ConfigFlowResult:
         """Configure origin by using an entity."""
         if user_input is not None:
             self._config[CONF_ORIGIN_ENTITY_ID] = user_input[CONF_ORIGIN_ENTITY_ID]
+            self._config.pop(CONF_ORIGIN_LATITUDE, None)
+            self._config.pop(CONF_ORIGIN_LONGITUDE, None)
```

Comment:
Add a comment explaining we pop these to clear previous configuration using coordinates